interview-questions.md

```diff
@@ -8,36 +8,47 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-Your answer:
+Your answer: You can fix this by associating the person with the Cohort. Since the Cohort has the primary key, as indicated by has_many students, we have to associate the student that belongs_to Cohort. The foreign key should be on the Student side. The foreign key can be added my creating a variable to store cohort and assign the cohort a student. The student should have a foreign key that matches the Cohort's cohort_id, which is their primary key.
 
-Researched answer:
+Researched answer:Since the Cohort model has the has_many description, students should follow after has_many, and this means the Cohort model will hold the primary key. The Student model which belongs_to the Cohort model will hold the foreign keys, as the foreign key always takes the side that has "belongs_to". To create the foreign key, we set cohort to a variable for each student and input their name, which will assign the foreingn key in the database for Student. For example, golf = cohort.first golf.students.create student: "Tabi Trubey". 
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: I don't think you HAVE to pass params to RESTful routes, as long as you're okay with hard coding everything and having static and generic code. However, I think the RESTful routes that must always be passed params are show, new, create, edit, update, and destroy because they are all operating on ONE specific item, which needs to be dynamic as it can change depending on what id is to be deleted or updated or created. I think index would be the only one that doesn't need params because it shows ALL of the data at all times. 
 
-Researched answer:
+Researched answer: The create and update RESTful routes must always be passed params. These params are called strong params and protect the security, integrity, and validitiy of a database. Create and update are passed strong params so an unauthorized user or "hacker" will not be able to create malicious data in the database or delete vital instances in the database. The strong params are indicated by the private method, which is created as the last method in the controller, because anything below it is included in the method and only create and update are intended to be the sole arguments. These params place requirements on what is permitted and where and cannot be changed. 
```

Comment:
You are right in regards to strong params, however, Show, Update, New, and Destroy need an id param to be able to know which instance you are working with.